dataclasses_avroschema/utils.py

```diff
@@ -80,6 +80,9 @@ def standardize_custom_type(value: typing.Any) -> typing.Any:
         return tuple(standardize_custom_type(v) for v in value)
     elif issubclass(type(value), enum.Enum):
         return value.value
+    elif callable(getattr(value, "asdict", None)):
+        # object has asdict() method, it's most likely our model
+        return {k: standardize_custom_type(v) for k, v in value.asdict().items()}
```

Comment:
`standardize_custom_type` already has handling for `dict`s, so we can delegate the the handling of items to a recursive call: 
```suggestion
        return standardize_custom_type(value.asdict())
```

Reply:
makes sense but as @joaoe mentioned - pydantic classes have `dict()` method we should call. AvroBaseModel classes (which inherit from pydantic and therefore also make this condition) have `asdict()` which should be called instead as it handles recursive serialization properly.

Reply:
@marcosschroh I think what @michal-rogala is concerned about here is that `pydantic.BaseModel` doesn't have an `asdict` method. But from what I can tell, nesting `pydantic.BaseModel` subclasses that aren't `AvroModel` or `AvroBaseModel` subclasses isn't supported (raises a ValueError, type {native_type} is unknown), so I think it's safe to assume that if we're at the point where we're running `standardize_custom_type` and `is_pydantic_model` is True, that the value is an `AvroBaseModel` subclass.

What this may not handle is an AvroModel nested in an AvroBaseModel. @marcosschroh Would you expect that nesting to be valid?

With @michal-rogala's version that checks for `asdict` this test would pass:
```python
def test_mixed_type_nesting():
    @dataclasses.dataclass
    class Friend(AvroModel):
        name: str
        hobbies: typing.List[str]

    class User(AvroBaseModel):
        name: str
        friends: typing.List[Friend]

        class Config:
            arbitrary_types_allowed = True

    user = User(name="Alex", friends=[Friend(name="Mr. Robot", hobbies=["fishing", "codding"])])

    assert User.deserialize(user.serialize()) == user
```

This fails using the `is_pydantic_model` version, because the nested `AvroModel` doesn't get converted. Another option is to check if `value` is a subclass of `AvroModel`, as `AvroBaseModel` is also a subclass of `AvroModel` and both define an `asdict` method:

```python
def standardize_custom_type(value: typing.Any) -> typing.Any:
    # Prevent circular import
    from .schema_generator import AvroModel

    if isinstance(value, dict):
        return {k: standardize_custom_type(v) for k, v in value.items()}
    elif isinstance(value, list):
        return [standardize_custom_type(v) for v in value]
    elif isinstance(value, tuple):
        return tuple(standardize_custom_type(v) for v in value)
    elif isinstance(value, AvroModel):
        return value.asdict()
    elif issubclass(type(value), enum.Enum):
        return value.value
    return value
```

Reply:
End users will always use either `dataclasses` (AvroModel) or `pydantic` (AvroBaseModel) but never a combination on both because the behavior is not the same and pydantic has extra functionalities that  `dataclasses` does not.

Indeed, it is safe to assume that `is_pydantic_model` will return  `True` for `AvroBaseModel`.

Reply:
so - to sum up, are we ok with checking with `is_pydantic_model` and call `asdict()`? 

The only question is: would it be possible to create an `AvroBaseModel` but nest a native pydantic BaseModel inside? In that case we would call `asdict()` on native BaseModel which would fail.

Reply:
No, that is not possible. The reason is that `pydantic BaseModel ` do not have the ability to create avro schemas, serialize and deserialize data in avro format. End users always must to use `AvroBaseModel` in all the models.

Reply:
> End users always must to use AvroBaseModel in all the models.
Is that enforced by the code elsewhere or are you hoping users don't break the API ?

Reply:
> yes, but how about this:
> 
> ```
> class Guest(BaseModel):
>    name: str
>    surname: str
>    
> class GuestBook(AvroBaseModel):
>   guests: list[Guest]
> ```
> 
> If I understand - it won't be able to generate record schema for `guests` field?

Exactly, and it will fail as `Guest` is an `unknown` type. It should be an `AvroBaseModel` which is translated to a `record` type in `avro`. The example that you provided will rause the excaption: `ValueError: Type <class '__main__.Guest'> is unknown. Please check the valid types at https://marcosschroh.github.io/dataclasses-avroschema/fields_specification/#avro-field-and-python-types-summary`